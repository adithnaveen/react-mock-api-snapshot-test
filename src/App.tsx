import React from 'react';
import TableList from './TableList'
import './App.css'
import axios from 'axios'

function App() {
  const [users, setUsers] = React.useState([])

  React.useEffect(() => {
    axios
      .get(`https://jsonplaceholder.typicode.com/users`)
      .then((res) => {
        setUsers(res.data)
      })
  }, [])

  return (
    <div className="App">
      <TableList data={users} />
    </div>
  )
}

export default App
