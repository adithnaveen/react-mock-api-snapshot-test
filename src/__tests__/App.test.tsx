import renderer, { act } from 'react-test-renderer'
import App from '../App'

const mockData = {
  data: [
    { name: 'a fake name', surname: 'fake surname', email: 'fake@email.com' },
    {
      name: 'a fake name 2',
      surname: 'fake surname 2',
      email: 'fake@email.com',
    },
  ],
}

jest.mock('axios', () => {
  return {
    get: jest.fn().mockImplementationOnce(() => Promise.resolve(mockData)),
  }
})

describe('App component', () => {
  it('it works', async () => {
    let tree
    await act(async () => {
      tree = renderer.create(<App />)
    })
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
