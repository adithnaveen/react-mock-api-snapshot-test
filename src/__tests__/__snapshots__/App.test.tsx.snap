// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component it works 1`] = `
<div
  className="App"
>
  <table
    className="sc-bczRLJ cZNNyj"
  >
    <tr>
      <th
        className="sc-gsnTZi dnOtss"
      >
        Name
      </th>
      <th
        className="sc-gsnTZi dnOtss"
      >
        Surname
      </th>
      <th
        className="sc-gsnTZi dnOtss"
      >
        Email
      </th>
    </tr>
    <tr>
      <td>
        a fake name
      </td>
      <td>
        fake surname
      </td>
      <td>
        fake@email.com
      </td>
    </tr>
    <tr>
      <td>
        a fake name 2
      </td>
      <td>
        fake surname 2
      </td>
      <td>
        fake@email.com
      </td>
    </tr>
  </table>
</div>
`;
